packages/core/src/modules/credentials/repository/CredentialExchangeRecord.ts

```diff
@@ -91,24 +90,22 @@ export class CredentialExchangeRecord extends BaseRecord<
       this.linkedAttachments = props.linkedAttachments
       this.revocationNotification = props.revocationNotification
       this.errorMessage = props.errorMessage
-      this.credentials = props.credentials ?? []
+      this.credentials = props.credentials || []
     }
   }
 
   public getTags() {
     const metadata = this.metadata.get(CredentialMetadataKeys.IndyCredential)
-    let credentialIds: string[] = []
-
+    let ids: string[] = []
     if (this.credentials) {
-      credentialIds = this.credentials.map((c) => c.credentialRecordId)
+      ids = this.credentials.map((c) => c.credentialRecordId)
     }
-
```

Comment:
With the change to `|| []` I think we can just do:

```ts
const credentialIds = this.credentials.map(c => c.credentialRecordId)
```

Reply:
That change doesn't work
Test fails

Reply:
Then we should look at why it is failing. The types indicate the array is always defined, so somewhere it's assigning an undefined value where it should be an empty array I think. Please investigate.